modules/core/src/test/scala/com/snowplowanalytics/iglu/schemaddl/parquet/MigrationSpec.scala

```diff
@@ -590,10 +521,153 @@ class MigrationSpec extends org.specs2.Specification {
     ) should beRight(true)
 
   }
+
+  // Preserve ordering of fields in a struct $e17
+  def e17 = {
+    val struct1 = Type.Struct(
+      NonEmptyVector.of(
+        Field("zzz", Type.String, Nullable),
+        Field("yyy", Type.String, Nullable),
+        Field("xxx", Type.String, Nullable),
+      )
+    )
+
+    val struct2 = Type.Struct(
+      NonEmptyVector.of(
+        Field("xxx", Type.String, Required),
+        Field("new1", Type.String, Required),
+        Field("yyy", Type.String, Required),
+        Field("new2", Type.String, Required),
```

Comment:
Could be nice to switch `new1` and `new2`

Reply:
If I switch `new1` and `new2` in the input, then I'll also need to swtich `new1` and `new2` in the expected output.  It's because the alphabetical ordering only happens when we call `Field.normalize`, and that is not part of this test.

You have made me realize though that this test is not realistic!  Because realistically we always call `normalize` first, and `mergeSchemas` afterwards.  So to make it realistic, `struct1` should start with fields `vvv`, `xxx`, `zzz`, and the new added fields should be `www` and `yyy`.

Reply:
> If I switch new1 and new2 in the input, then I'll also need to swtich new1 and new2 in the expected output

Yes this is what I meant, to make sure that the "wrong" alphabetical order is preserved.

> and that is not part of this test

Alright !

> Because realistically we always call normalize first, and mergeSchemas afterwards

I'm very glad that you say that ! I was wondering in which case we were calling `normalize` and in each case we were not.  Shouldn't `mergeSchemas` be in charge of calling `normalize` then?

Reply:
> Shouldn't `mergeSchemas` be in charge of calling `normalize` then?

Yeah that's a really good point.

Reply:
@benjben I pushed a commit in which `mergeSchemas` is in change of calling `normalize`.

...but unfortunately I think I need to revert it.  It's because of [this comment](https://github.com/snowplow-incubator/common-streams/blob/0.8.1/modules/loaders-common/src/main/scala/com/snowplowanalytics/snowplow/loaders/transform/TypedTabledEntity.scala#L105-L106)

> Must normalize first and add the schema key field after. To avoid unlikely weird issues with similar existing keys.

To be strictly correct under all circumstances, the order of operations must be:

1. Normalize the field
2. Add the `_schema_version` key.
3. Merge the fields.

While it is _probably_ ok to mix up that order for most schemas, it is not guaranteed to be ok for all possible schemas.  And I cannot think of a clever way to keep that order, while also making `mergeSchemas` responsible for calling `normalize`.

So on balance I think it is better to say we always normalize first and merge afterwards.

Maybe we can improve on this in future -- but this is stretching beyond what I'm trying to address in this current PR.

Reply:
Couldn't `addSchemaVersion` just be a `boolean` parameter of `mergeSchemas` ?

Otherwise that's fine to leave it as is for now, at least we have considered the options.

Reply:
It could yeah... [this is the logic](https://github.com/snowplow-incubator/common-streams/blob/0.8.1/modules/loaders-common/src/main/scala/com/snowplowanalytics/snowplow/loaders/transform/TypedTabledEntity.scala#L119-L133) that we would have to migrate from common-streams over to schema-ddl.  But it's a bit awkward because there is some stuff there which is "snowplow-aware" -- meaning, how we add the `_schema_version` key to objects within a contexts array.  Whereas schema-ddl currently doesn't know about snowplow details like how contexts are arrays.

Plus if we make any change to `mergeSchemas` then we need to carry it over to rdb-loader, which is a bit fragile at the moment.

I do like the ideas to tidy-up the inter-dependencies between two libraries (schema-ddl and common-streams), but I'm going to skip it for this PR.